Admin/Adapters/FileSystem/FileSystemDal.ts

```diff
@@ -0,0 +1,60 @@
+import fs = require('fs');
+import Port = require('../../Ports/IDal');
+import path = require('path');
+
+class FileSystemDal implements Port.IDal {
+  /**
+   * Find all files inside a dir, recursively.
+   * @function getAllFilesInDirectoryRecursive
+   * @param  {string} dir Dir path string.
+   * @return {string[]} Array with all file names that are inside the directory.
+   */
+  public getAllFilesInDirectoryRecursive = (dir: string): string[] =>
+    fs.readdirSync(dir).reduce((files: string[], file: string): string[] => {
+      const name = path.join(dir, file);
+      const isDirectory = fs.statSync(name).isDirectory();
+      return isDirectory
+        ? [...files, ...this.getAllFilesInDirectoryRecursive(name)]
+        : [...files, name];
+    }, []);
+
+  /**
+   * Returns all requests types
+   */
+  public getAllRequests(): string[] {
+    console.log(`Getting requests...`);
+    return JSON.parse(
+      fs.readFileSync('../Data/Requests/RequestsConfig.json', 'utf8')
```

Comment:
This should be injected to the FileSystemDal as the Requests config path

Reply:
Agree.
inject within the c'tor?

Reply:
It's kind of problemtic since it is implementing an interface which not always need those path params, it is only in case of file system